### ISSUE TITLE
FetchError: invalid json response body  reason: Unexpected token < in JSON at position 0

### ISSUE THREAD

**Issue by Procuria** [davidlday/vscode-languagetool-linter#286]
Hello,
thank you for the work to make this plugin possible.

I got a self-hosted LanguageTool-Instance running and working properly. 
Specs: 
- dockerized
- behind nginx-reverse proxy

When I try to lint texts in VSCode I got this error message:

```
Error connecting to https://url.selfhosted.languagetool.tld:443/v2/check
FetchError: invalid json response body at https://url.selfhosted.languagetool.tld:443/v2/check reason: Unexpected token < in JSON at position 0
``` 
I checked the logs of my LanguageToolInstance - no requests are arriving there.
I set the domain to trusted in VSCode.
I tried to set the url in the VSCode config with and without ports.

Could you please point me the direction what could be wrong?

**Comment by davidlday** [comment]
Usually when I see this it's because the server is returning an error instead of the expected JSON. Can you test your server using curl to verify it's responding correctly? Please run this at the command prompt and share the response:

```bash
curl -X POST --header 'Content-Type: application/x-www-form-urlencoded' \
   --header 'Accept: application/json' \
   -d 'text=This%20is%20a%20test.&language=en-US&enabledOnly=false' \
   'https://url.selfhosted.languagetool.tld:443/v2/check'
```

**Comment by Procuria** [comment]
@davidlday Thank you very much for the response. Indeed, I get an error using the recommended request (ironically the text I write here is corrected flawlessly by the same languagetool-server via the browser-plugin).
So should I check my nginx config regarding headers etc. ?   
````
<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx/1.19.10</center>
</body>
</html>
````

**Comment by davidlday** [comment]
Yes, check your nginx config. It doesn't seem to be passing the call on to LanguageTool, hence the 404. I'm no expert on nginx, but if you share your config I might be able to help.

**Comment by Procuria** [comment]
@davidlday thank you for your kind offer and your support. I was able to spot the error, or better lack of configuration regarding your great plugin, on my side. I briefly explain, so maybe it can help others too.
My base NGINX proxy_pass config, which worked for the native Browser-Plugins flawlessly, was:
````
location = /check{
		proxy_pass http://127.0.0.1:8010/v2/check;
	}
```` 

I checked your code and discovered [here](https://github.com/davidlday/vscode-languagetool-linter/blob/7ea7b44c88b06ad06165391f21eb657b4e0f53c0/src/Constants.ts#L60) that the path `/v2/check` is 'hardcoded' for all request. No wonder that I got a 404, as my NGINX config lacks this location-block.
So I added:
````
location = /v2/check{
		proxy_pass http://127.0.0.1:8010/v2/check;
	}
```` 
Taaadaaa - problem solved. 🥳 Now my readme-files will be much more readable 🎆 
Thank you for your work on this and the possibility to learn!
